enterprise/views.py

```diff
@@ -1453,7 +1454,7 @@ def get(self, request, *args, **kwargs):
 
         # Directly enroll in audit mode if the request in question has full direct audit enrollment eligibility.
         resource_id = course_run_id or program_uuid
```

Comment:
I think if you change this to

```
resource_id = course_run_id or program_uuid or course_key
```

then no other code change here is needed.

Reply:
Nevermind, this may not work for the actual `enroll` function or similar right below it though.